packages/runtime/src/widgets/Form/Radio.tsx

```diff
@@ -21,20 +24,50 @@ export const RadioWidget: React.FC<RadioWidgetProps> = (props) => {
     props.id,
     { setValue },
   );
+  const action = useEnsembleAction(props.onChange);
+
+  const onChangeCallback = useCallback(
+    (newValue: string) => {
+      if (!action) {
+        return;
+      }
+      action.callback({ value: newValue });
+    },
+    [action],
+  );
+
+  const handleChange = (newValue: string): void => {
+    setValue(newValue);
+    onChangeCallback(newValue);
+  };
 
   return (
     <EnsembleFormItem values={values}>
       <Radio.Group
         disabled={
           values?.enabled === undefined ? false : Boolean(values.enabled)
         }
-        onChange={(event): void => setValue(String(event.target.value))}
+        onChange={(event): void => handleChange(String(event.target.value))}
         ref={rootRef}
-        style={values?.styles}
+        style={{
+          ...values?.styles,
+          ...(values?.styles?.visible === false
+            ? { display: "none" }
+            : undefined),
```

Comment:
This is taken care of in the parent form item component - it shouldn't need to be repeated